lib/ood_core/app_config.rb

```diff
@@ -0,0 +1,168 @@
+
+        def enable_native_vnc
+            !! raw_config['enable_native_vnc']
+        end
+        alias_method :enable_native_vnc?, :enable_native_vnc
+
+        def file_upload_max
+            (raw_config['file_upload_max'] || 10485760000).to_i
+        end
+
+        def motd_format
+            raw_config.dig('motd', 'format')
+        end
+
+        def motd_uri
+            raw_config.dig('motd', 'uri')
+        end
+
+        def announcement_path
+            Array.wrap(raw_config['announcement_path'] || []).map do |path|
+                Pathname.new(path)
+            end
+        end
+
+        def app_catalog_url
+            raw_config['app_catalog_url']
+        end
+
+        # Replaces: app_development
+        def show_app_development
+            !! raw_config['app_development']
+        end
+        alias_method :show_app_development?, :show_app_development
+
+        def app_sharing
+            !! raw_config['app_sharing']
+        end
+        alias_method :app_sharing?, :app_sharing
+
+        def dev_ssh_host
+            raw_config['dev_ssh_host']
+        end
+
+        def user_file_system_locations
+            Array.wrap(raw_config['user_file_system_locations'] || []).map do |location|
+                Pathname.new(location)
+            end
+        end
+
+        def links
+            raw_config['links'] || {}
+        end
+
+        def load_external_bc_config
+            setting = raw_config['load_external_bc_config']
+            return true if setting.nil?
+
+            !! setting
+        end
+        alias_method :load_external_bc_config?, :load_external_bc_config
+
+        def load_external_config
+            setting = raw_config['load_external_config']
+            return true if setting.nil?
+
+            !! setting
+        end
+        alias_method :load_external_config?, :load_external_config
+
+        def locale
+            raw_config['locale'] || 'en'
+        end
+
+        def locales_root
+            Pathname.new(raw_config['locales_root'] || '/etc/ood/config/locales')
+        end
+
+        def quota_path
+            Array.wrap(raw_config['quota_path'] || []).map do |path|
+                Pathname.new(path)
+            end
+        end
+
+        def ssh_hosts
+            raw_config['ssh_hosts'] || []
+        end
+
+        def poll_delay
+            (raw_config['poll_delay'] || 10000).to_i
+        end
+
+        def show_all_apps_link
+            !! raw_config['show_all_apps_link']
+        end
+        alias_method :show_all_apps_link?, :show_all_apps_link
+
+        def whitelist_path
+            raw_config['whitelist_path'] || []
+        end
+
+        class Error < StandardError; end
+
+        private
+
+        def load_yaml
+            erb_wrapper_path = Pathname.new(ENV['OOD_CONFIG_ERB_WRAPPER'] || '/etc/ood/config/bin/erb_wrapper')
+            app_config_path = Pathname.new(ENV['OOD_APP_CONFIG'] || '/etc/ood/config/app_config.yml')
```

Comment:
> If we are supporting ERB as suggested in https://github.com/OSC/ood_core/pull/154, should we add the ERB here to build the list of favorites?

_Originally posted by @ericfranz in https://github.com/OSC/osc-ood-config/pull/39

Note that it is really only suggested to use ERB. So long as it is executable (and works in the calling environment) and outputs YAML to stdout there is no reason it couldn't be written in another language.

This section needs more work. There needs to be a fallback if neither `OOD_CONFIG_ERB_WRAPPER` is set or `/etc/ood/config/bin/erb_wrapper` does not exist.